linker.go

```diff
@@ -0,0 +1,253 @@
+				if err != nil {
+					t.Error(err)
+				}
+			}()
+
+			Convey("The error should be nil", func() {
+				So(err, ShouldBeNil)
+			})
+
+			Convey("The object mode should be link", func() {
+				// Link object's mode must be link.
+				So(o.Mode.IsLink(), ShouldBeTrue)
+			})
+
+			Convey("The linkTarget of the object must be the same as the target", func() {
+				// The linkTarget must be the same as the target.
+				linkTarget, ok := o.GetLinkTarget()
+
+				So(ok, ShouldBeTrue)
+				So(linkTarget, ShouldEqual, target)
+			})
+
+			Convey("Stat should get path object without error", func() {
+				obj, err := store.Stat(path)
+
+				Convey("The error should be nil", func() {
+					So(err, ShouldBeNil)
+				})
+
+				Convey("The object mode should be link", func() {
+					// Link object's mode must be link.
+					So(obj.Mode.IsLink(), ShouldBeTrue)
+				})
+
+				Convey("The linkTarget of the object must be the same as the target", func() {
+					// The linkTarget must be the same as the target.
+					linkTarget, ok := obj.GetLinkTarget()
+
+					So(ok, ShouldBeTrue)
+					So(linkTarget, ShouldEqual, target)
+				})
+			})
+		})
+
+		Convey("When create a link object from a not existing target", func() {
+			target := uuid.New().String()
+
+			path := uuid.New().String()
+			o, err := l.CreateLink(path, target)
+
+			defer func() {
+				err = store.Delete(path)
+				if err != nil {
+					t.Error(err)
+				}
+			}()
+
+			Convey("The error should be nil", func() {
+				So(err, ShouldBeNil)
+			})
+
+			Convey("The object mode should be link", func() {
+				// Link object's mode must be link.
+				So(o.Mode.IsLink(), ShouldBeTrue)
+			})
+
+			Convey("The linkTarget of the object must be the same as the target", func() {
+				linkTarget, ok := o.GetLinkTarget()
+
+				So(ok, ShouldBeTrue)
+				So(linkTarget, ShouldEqual, target)
+			})
+
+			Convey("Stat should get path object without error", func() {
+				obj, err := store.Stat(path)
+
+				Convey("The error should be nil", func() {
+					So(err, ShouldBeNil)
+				})
+
+				Convey("The object mode should be link", func() {
+					// Link object's mode must be link.
+					So(obj.Mode.IsLink(), ShouldBeTrue)
+				})
+
+				Convey("The linkTarget of the object must be the same as the target", func() {
+					// The linkTarget must be the same as the target.
+					linkTarget, ok := obj.GetLinkTarget()
+
+					So(ok, ShouldBeTrue)
+					So(linkTarget, ShouldEqual, target)
+				})
+			})
+
+			size := rand.Int63n(4 * 1024 * 1024) // Max file size is 4MB
+			content, _ := ioutil.ReadAll(io.LimitReader(randbytes.NewRand(), size))
+
+			_, err = store.Write(target, bytes.NewReader(content), size)
+			if err != nil {
+				t.Fatal(err)
+			}
+
+			defer func() {
+				err = store.Delete(target)
+				if err != nil {
+					t.Error(err)
+				}
+			}()
+
+			Convey("Read should get path object data without error", func() {
+				var buf bytes.Buffer
+				n, err := store.Read(path, &buf)
```

Comment:
`LinkObject` is not readable, so we can't read from a link directly. Instead, we need to read from the link object's `LinkTarget`.

Maybe we can remove tests about `Read should get path object data without error`?

Reply:
Got it!